src/symbol/placement.js

```diff
@@ -279,12 +280,15 @@ class Placement {
             const isDuplicate = seenCrossTileIDs[symbolInstance.crossTileID];
 
             let opacityState = this.opacities[symbolInstance.crossTileID];
-            if (!opacityState) {
+            if (isDuplicate) {
+                opacityState = duplicateOpacityState;
+            } else if (!opacityState) {
                 opacityState = defaultOpacityState;
+            }
+
+            if (!this.opacities[symbolInstance.crossTileID]) {
```

Comment:
Can't we just move this into the `else if (!opacityState)` clause above and avoid a second lookup? Because it shouldn't be possible for us to have gone down the `isDuplicate` branch without having first made an entry for this `this.opacities[crossTileID]` (and if it were possible, it'd be a problem, because we'd be making our initial entry into `duplicateOpacityState` instead of `defaultOpacityState`).

Reply:
Yep! fixed